### ISSUE TITLE
Serializing Timestamps/Dates embedded in a "$date" field

### ISSUE THREAD

**Issue by gschmutz** [an0r0c/kafka-connect-transform-tojsonstring#10]
First thanks for implementing this SMT!

Have just tried to use it for serializing an Avro message to an Oracle table as JSON and timestmap as well as time and date types are serialized embedded in a `$date` field as shown here (I'm using the default RELAXED output mode:

```json
{"order": 
  { "guid": "cd38f167-37bd-494a-b065-fb35133b1e7b", 
    "id": 0, "orderDate": {"$date": "2022-12-03T10:22:28.614Z"}, 
    "localOrderDate": 1670066548616, "customerId": 2, 
    "orderLines": [{"lineId": 1, "productId": 12, "quantity": 10}, {"lineId": 2, "productId": 20, "quantity": 5}, {"lineId": 3, "productId": 10, "quantity": 1}], 
    "createdDate": {"$date": "2022-12-03T00:00:00Z"}, 
    "createdTime": {"$date": "1970-01-01T11:22:28.616Z"}
 }
}
```

The problem with that is, that I couldn't figure out a way to access it in Oracle using the `JSON_TABLE` function. It would be better to have it serialized as `"orderDate": "2022-12-03T10:22:28.614Z"`. 

Found a way to change the conversion to date by specifying a custom converter: <https://www.baeldung.com/java-convert-bson-to-json#custom-bson-to-json-date-conversion>.  

Have implemented it as a proof-of-concept locally and it works. 
Would you be interested to include it in your SMT? If yes, I can provide it as a pull request.

**Comment by gschmutz** [comment]
FYI: just realized that the `$date` embedding is related to using the logical types in Avro for dates and timestamps. Will create a separate Pull Request with the `Record2JsonStringConverterTest` unit test enhanced with logical types.

**Comment by an0r0c** [comment]
Hi! 

I never faced this issue before but I think it's because we didn't use the logical Types in Avro we just used epoch timestamps without any further metadata and therefore it was also written to the JSON as INT64.

One alternative might be to chain the TimestampConverter (https://docs.confluent.io/platform/current/connect/transforms/timestampconverter.html#properties)  before the tojsonstring smt.  So you can transform the avro epoch to a datetime string before the transformation to JSON is done and it is transformed to json as normal string. 

If you have a PR that adds this configurable to the SMT I'm happy to merge it.

**Comment by gschmutz** [comment]
Hi

Thanks for your repsonse. You are right, using the TimestampConverter before your SMT should work. Will try that as well and then decide if it would still make sense to include similar functionality in your SMT. 

Have just created a PR for the enhancement of the unit test to include the logical types.

**Comment by an0r0c** [comment]
Thanks for the PR with the unit test enhancement. I'll close this issue. If you have more PRs feel free ;)

**Comment by gschmutz** [comment]
@an0r0c  thanks for merging, just sent you a PR with the changes to support handling the logical types. Tried to use the TimestampConverter yesterday, but it does not work for me, as my date fields are nested inside a nested Avro record (nested object) and using SMT for nested fields are not yet supported (found that there is [KIP-821](https://cwiki.apache.org/confluence/display/KAFKA/KIP-821%3A+Connect+Transforms+support+for+nested+structures) to support that in the future). That's why I decided to add the logic to your SMT. Just let me know if you have any questions or comments on the implementation.
